### PR TITLE
v4l2 support in westeros-sink

### PR DESCRIPTION
**Pull request by Khan3033**

### CODE REVIEW

File: recipes-core/packagegroups/packagegroup-westeros.bb [hunk 1]
```diff
@@ -25,4 +25,4 @@ WESTEROS_SOC_rpi = "westeros-soc-drm"
 WESTEROS_SOC_rpi = "${@bb.utils.contains("MACHINE_FEATURES", "vc4graphics", "westeros-soc-drm", "westeros-soc-rpi", d)}"
 
 WESTEROS_RENDERER ?= ""
-WESTEROS_SOC_rpi = "${@bb.utils.contains("MACHINE_FEATURES", "vc4graphics", "", "westeros-render-dispmanx", d)}"
+#WESTEROS_SOC_rpi = "${@bb.utils.contains("MACHINE_FEATURES", "vc4graphics", "", "westeros-render-dispmanx", d)}"
```
Comment (psivasubramanian) [inline recipes-core/packagegroups/packagegroup-westeros.bb]: WESTEROS_SOC_rpi = "${@bb.utils.contains("MACHINE_FEATURES", "vc4graphics", "westeros-soc-drm", "westeros-render-dispmanx", d)}" should work for both vc4graphics and userland.

File: recipes-graphics/westeros/westeros.inc [hunk 2]
```diff
@@ -5,7 +5,7 @@ PV = "1.0+git${SRCPV}"
 SRC_URI = "${WESTEROS_URI}"
 SRCREV = "${WESTEROS_SRCREV}"
 WESTEROS_URI ?= "git://github.com/rdkcmf/westeros"
-WESTEROS_SRCREV ?= "af611caeb2d97f3f14950d7b724b47e53ec34b28"
+WESTEROS_SRCREV = "${AUTOREV}"
```
Comment (psivasubramanian) [inline recipes-graphics/westeros/westeros.inc]: Please use fixed revision only.

File: recipes-wpe/wpebackend-rdk/wpebackend-rdk_0.1.bb [hunk 3]
```diff
@@ -33,7 +32,7 @@ PACKAGECONFIG ?= "${WPE_BACKEND} virtualinput"
 PACKAGECONFIG[imx6]             = "-DUSE_BACKEND_VIV_IMX6_EGL=ON,,imx-gpu-viv"
 PACKAGECONFIG[intelce]          = "-DUSE_BACKEND_INTEL_CE=ON,,intelce-display"
 PACKAGECONFIG[nexus]            = "-DUSE_BACKEND_BCM_NEXUS=ON,,broadcom-refsw"
-PACKAGECONFIG[rpi]              = "-DUSE_BACKEND_BCM_RPI=ON,,userland"
+PACKAGECONFIG[rpi]              = "-DUSE_BACKEND_BCM_RPI=ON,,mesa"
```
Comment (psivasubramanian) [inline recipes-wpe/wpebackend-rdk/wpebackend-rdk_0.1.bb]: It's better to have vc4graphics as new PACKAGECONFIG and select based on MACHINE_FEATURES.

File: recipes-graphics/westeros/westeros.bbappend [hunk 4]
```diff
@@ -7,15 +7,16 @@ do_configure_prepend() {
 
 # bbappend for raspberryPi
 #
-PACKAGECONFIG_rpi = "incapp inctest increndergl incsbprotocol xdgv4"
-CXXFLAGS_append_rpi = " -DWESTEROS_PLATFORM_RPI -DWESTEROS_INVERTED_Y -DBUILD_WAYLAND -I${STAGING_INCDIR}/interface/vmcs_host/linux"
+PACKAGECONFIG_rpi = "incapp inctest incplayer increndergl incsbprotocol xdgv4"
+CXXFLAGS_append_rpi = "${@bb.utils.contains('MACHINE_FEATURES', 'vc4graphics', 'null', ' -DWESTEROS_PLATFORM_RPI -DWESTEROS_INVERTED_Y -DBUILD_WAYLAND -I${STAGING_INCDIR}/interface/vmcs_host/linux', d)}"
+CXXFLAGS_remove_rpi = "${@bb.utils.contains("MACHINE_FEATURES", "vc4graphics", "-DWESTEROS_PLATFORM_RPI -DWESTEROS_INVERTED_Y -I${STAGING_INCDIR}/interface/vmcs_host/linux", "", d)}"
```
Comment (psivasubramanian) [inline recipes-graphics/westeros/westeros.bbappend]: CXXFLAGS append and remove are contracting here.

File: recipes-core/images/wpe-image.inc [hunk 5]
```diff
@@ -1,6 +1,8 @@
 IMAGE_INSTALL_append = "\
    packagegroup-wpewebkit \
    packagegroup-wpeframework \
+   packagegroup-westeros \
```
[the remark this answers is not in the document]
Comment (wouterlucas) [reply recipes-core/images/wpe-image.inc]: Agree, not all images require westeros. I prefer the other approach when Wayland is included.

**Comment by wouterlucas** [comment]
Is this PR superseded by https://github.com/WebPlatformForEmbedded/meta-wpe/pull/295 ?
If so please close.

**Comment by psivasubramanian** [comment]
Yes @wouterlucas,  this can be closed.

**Pull request closed without merging**